examples/platform/efr32/rs911x/hal/efx_spi.c

```diff
@@ -287,10 +287,24 @@ int16_t rsi_spi_transfer(uint8_t * tx_buf, uint8_t * rx_buf, uint16_t xlen, uint
         /*
          * Wait for the call-back to complete
          */
-        if (xSemaphoreTake(spi_sem, portMAX_DELAY) == pdTRUE)
+        if (xSemaphoreTake(spi_sem, pdMS_TO_TICKS(RSI_SEM_BLOCK_MIN_TICK_VALUE)) == pdTRUE)
```

Comment:
The logic around the semaphore seems bad. 

How and why are you taking a semaphore within a 50 ms delay, and if you succeed, you give it back immediately ?

And if the Semaphore is held by someone else, you do SPI operation and then try to take it again ? And this time you're willing to wait for an infinite amount of time just to give it back immediately again ?

The whole logic needs rework, or at least a couple of comments...

Reply:
Looking at it, above we do a take and I believe we expect the 2nd take to block until the callback gives it back. So effectively this is a barrier to wait for the callback. The comment above seems to try to explain it, but just says "callback to complete" without saying callback gives back the semaphoe. The logic seems ok though.